tests/test_cli.py

```diff
@@ -15,18 +20,317 @@ def test_header(mock_cli):
+            "id": "idgui",
+            "command-only": None,
+            "params-out": "/path/params/out",
+            "params-in": temp_params_in.name,
+            "save-all": None,
+            "show-hidden": None,
+            "url": "builder_url",
+        }
+        cmd = ["launch"] + self.assemble_params(params) + ["pipeline_name"]
+        result = self.invoke_cli(cmd)
+
+        assert result.exit_code == 0
+
+        mock_launcher.assert_called_once_with(
+            cmd[-1],
+            params["revision"],
+            "command-only" in params,
+            params["params-in"],
+            params["params-out"],
+            "save-all" in params,
+            "show-hidden" in params,
+            params["url"],
+            params["id"],
+        )
+
+        mock_launcher.return_value.launch_pipeline.assert_called_once()
+
+    @mock.patch("nf_core.launch.Launch")
+    def test_cli_launch_no_params_in(self, mock_launcher):
+        """Test nf-core pipeline fails when params-in does not exist"""
+        mock_launcher.return_value.launch_pipeline.return_value = True
+
+        params = {
+            "params-in": "/fake/path",
+        }
+        cmd = ["launch"] + self.assemble_params(params) + ["pipeline_name"]
+        result = self.invoke_cli(cmd)
+
+        assert result.exit_code == 2
+        assert f"Invalid value for '-p' / '--params-in': Path '{params['params-in']}' does not exist." in result.output
+
+        mock_launcher.assert_not_called()
+
+    @mock.patch("nf_core.launch.Launch")
+    def test_cli_launch_fail(self, mock_launcher):
+        """Test nf-core pipeline fails with exit code 1 when pipeline fails."""
+        mock_launcher.return_value.launch_pipeline.return_value = False
+        cmd = ["launch", "pipeline_name"]
+        result = self.invoke_cli(cmd)
+        assert result.exit_code == 1
+
+    @mock.patch("nf_core.download.DownloadWorkflow")
+    def test_cli_download(self, mock_dl):
+        """Test nf-core pipeline is downloaded and cli parameters are passed on."""
+        params = {
+            "revision": "abcdef",
+            "outdir": "/path/outdir",
+            "compress": "tar.gz",
+            "force": None,
+            "container": "singularity",
+            "singularity-cache-only": None,
+            "parallel-downloads": 2,
+        }
+
+        cmd = ["download"] + self.assemble_params(params) + ["pipeline_name"]
+        result = self.invoke_cli(cmd)
+
+        assert result.exit_code == 0
+
+        mock_dl.assert_called_once_with(
+            cmd[-1],
+            params["revision"],
+            params["outdir"],
+            params["compress"],
+            "force" in params,
+            params["container"],
+            "singularity-cache-only" in params,
+            params["parallel-downloads"],
+        )
+
+        mock_dl.return_value.download_workflow.assert_called_once()
+
+    @mock.patch("nf_core.licences.WorkflowLicences")
+    def test_licences(self, mock_lic):
```

Comment:
this command is not working for dsl2 pipelines and we should deprecate it soon.

Reply:
Right, I'll add a deprecation warning then, I guess the test can stay for now